container-engine-lib/lib/backend_impls/docker/docker_manager/docker_auth.go

```diff
@@ -206,7 +209,41 @@ func GetAuthFromDockerConfig(repo string) (*registry.AuthConfig, error) {
 
 	// 3. Fallback to credentials in "auths" if no credStore is available
 	if auth, exists := authConfig.Auths[registryHost]; exists {
-		return &auth, nil
+		// Apply '/' to the end of the registry host if it doesn't have it
+		if !strings.HasSuffix(registryHost, "/") {
+			registryHost = registryHost + "/"
+		}
+		ac := registry.AuthConfig{
+			ServerAddress: registryHost,
+			Username:      auth.Username,
+			Password:      auth.Password,
+			Auth:          auth.Auth,
+			Email:         auth.Email,
+			IdentityToken: auth.IdentityToken,
+			RegistryToken: auth.RegistryToken,
+		}
+
+		// If the base64 encoded auth field is set, decode it and also set the Username and Password
+		if auth.Auth != "" {
+			// Base64 decode it and set the Username and Password
+			decodedAuth, err := base64.StdEncoding.DecodeString(auth.Auth)
+			if err != nil {
+				return nil, stacktrace.Propagate(err, "error decoding auth for registry '%s'", registryHost)
+			}
+			ac.Username = string(decodedAuth[:strings.IndexByte(string(decodedAuth), ':')])
+			ac.Password = string(decodedAuth[strings.IndexByte(string(decodedAuth), ':')+1:])
+			ac.Auth = auth.Auth
+		}
+
+		// If the username or password fields are set, set them in the AuthConfig (Overrides the decoded auth)
+		if auth.Username != "" {
+			ac.Username = auth.Username
+		}
+		if auth.Password != "" {
+			ac.Password = auth.Password
+		}
```

Comment:
unlikely, but is it ever possible to run into a case where `Username` is non empty but `Password` is empty? in which case you'd have the `Password` from decoded auth and the `Username` from `auth`? Probably unlikely but maybe worth adding a test for